scripts/set_commit_hook.sh

```diff
@@ -78,7 +78,24 @@ fi
 if [[ \"\$whitespace_present\" == \"yes\" || \"\$tab_present\" == \"yes\" ]]; then
   exit 1
 fi
-git diff-index --check --cached HEAD -- ':(exclude)*/reference/*' ':(exclude)*.patch'"
+git diff-index --check --cached HEAD -- ':(exclude)*/reference/*' ':(exclude)*.patch'
+
+check_format_file() {
+  file=\$1
```

Comment:
I would really prefer to have clang format applied before compilation as this check can be really annoying

Reply:
I would do both